linkerd/proxy/core/src/listen.rs

```diff
@@ -118,10 +127,14 @@ where
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         loop {
-            try_ready!(self.accept.poll_ready());
+            try_ready!(self.accept.poll_ready().map_err(Into::into));
             let conn = try_ready!(self.listen.poll_accept().map_err(Into::into));
-            let accept = self.accept.accept(conn).map_err(|e| match e {});
-            self.executor.spawn(Box::new(accept)).map_err(Error::from)?;
+            let accept = self.accept.accept(conn).map_err(|_| {});
+            self.executor
+                .spawn(Box::new(
+                    accept.and_then(|conn_future| conn_future.map_err(|_| {})),
```

Comment:
If I understand correctly, this just silently drops errors? I think we need to restore the `Error = Never` constraints (for both error types) and modify accept_error to handle both.

Reply:
Yeah, IIRC, when we were pairing on this, we noted that the `accept_error`  layer should be updated to handle both future's errors?

Reply:
makes total sense. Just to solidify my understanding a `.map_err(|e| match e {})` when `Error = Never` works and does not complain about non exhaustive matches because the compiler knows that Never cannot be instantiated. Correct? And I assume this further serves as a proove that this future can actually never error. This is quite neat.

Reply:
That's right. Never is just `enum Never {}` with some implementations for Error, Display, etc.